tsdb/head.go

```diff
@@ -1231,7 +1243,15 @@ func (a *headAppender) Commit() error {
 	return nil
 }
 
-func (a *headAppender) Rollback() error {
+func (a *headAppender) Rollback() (err error) {
+	if a.closed {
+		return ErrAppenderClosed
+	}
+	defer func() {
+		if err == nil {
+			a.closed = true
```

Comment:
I think we should set this even if `err != nil`. We also do all the other deferred stuff below without condition on the error. Calling `Rollback()` twice would probably wreak havoc in any case.